source/common/http/filter/fault_filter.cc

```diff
@@ -208,9 +211,15 @@ void FaultFilter::recordAbortsInjectedStats() {
   config_->stats().aborts_injected_.inc();
 }
 
-FilterDataStatus FaultFilter::decodeData(Buffer::Instance&, bool) {
-  return delay_timer_ == nullptr ? FilterDataStatus::Continue
-                                 : FilterDataStatus::StopIterationAndBuffer;
+FilterDataStatus FaultFilter::decodeData(Buffer::Instance& data, bool) {
+  if (delay_timer_ == nullptr) {
+    return FilterDataStatus::Continue;
+  }
+  if (buffer_limit_ > 0 && data.length() > buffer_limit_ && !high_watermark_called_) {
```

Comment:
In this case, perhaps filter can somehow tell connection manager whether to reset and return 413 vs. fire callbacks?